libyul/backends/wasm/WasmDialect.cpp

```diff
@@ -56,34 +57,53 @@ WasmDialect::WasmDialect()
 	m_functions["i64.eq"_yulstring].returns.front() = "i32"_yulstring;
 	m_functions["i64.ne"_yulstring].returns.front() = "i32"_yulstring;
 
-	addFunction("i64.eqz", 1, 1);
+	addFunction("i32.eqz", "i32"_yulstring, 1, 1);
```

Comment:
```suggestion
	addFunction("eqz", "i32"_yulstring, 1, 1);
```
Shouldn't it be like this?

Reply:
Why?

Reply:
I think this is related to me still not understanding what https://github.com/ethereum/solidity/pull/8161/files?file-filters%5B%5D=.h#diff-6311061a591e248cfc780cfd84bd3f22R61 does

Reply:
`i32.eqz` is the name of the wasm function. Some types, but not all, are part of the names of wasm functions. The parameter you linked is the type applied to all function parameters and return variables.

Reply:
Ah ok